code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -213,7 +213,7 @@
 		changed_name = custom_name
 	if(changed_name == "" && client)
 		changed_name = client.prefs.custom_names["cyborg"]
-	if(changed_name == "")
+	if(!changed_name || changed_name == "")
```

Comment:
@MrStonedOne tells me that old client prefs can have null for the cyborg name. So I think this is where the bug comes from.